rust-runtime/aws-smithy-types/src/date_time/de.rs

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use super::*;
+use serde::de::Visitor;
+use serde::Deserialize;
+
+struct DateTimeVisitor;
+
+enum VisitorState {
+    Second,
+    SubsecondNanos,
+    Unexpected,
```

Comment:
It looks like this unexpected state is never set, so it's probably not needed.

Reply:
It was meant to protect from something un-expected(it was set after subsec nano), but I think we don't need it.